// [[Reason: Invalid computed property can only be detected by TSC]] ////

//// [tests/cases/compiler/indexSignatureMustHaveTypeAnnotation.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,8 +1,9 @@
 
 
 //// [indexSignatureMustHaveTypeAnnotation.d.ts]
 interface I {
+    [x]: string;
     [x: string]: any;
 }
 declare class C {
     [x]: string;
@@ -16,15 +17,14 @@
 indexSignatureMustHaveTypeAnnotation.ts(3,5): error TS1169: A computed property name in an interface must refer to an expression whose type is a literal type or a 'unique symbol' type.
 indexSignatureMustHaveTypeAnnotation.ts(3,6): error TS2304: Cannot find name 'x'.
 indexSignatureMustHaveTypeAnnotation.ts(4,5): error TS1021: An index signature must have a type annotation.
 indexSignatureMustHaveTypeAnnotation.ts(9,5): error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-indexSignatureMustHaveTypeAnnotation.ts(9,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 indexSignatureMustHaveTypeAnnotation.ts(9,6): error TS2304: Cannot find name 'x'.
 indexSignatureMustHaveTypeAnnotation.ts(9,6): error TS4031: Public property '[x]' of exported class has or is using private name 'x'.
 indexSignatureMustHaveTypeAnnotation.ts(14,5): error TS1021: An index signature must have a type annotation.
 
 
-==== indexSignatureMustHaveTypeAnnotation.ts (8 errors) ====
+==== indexSignatureMustHaveTypeAnnotation.ts (7 errors) ====
     interface I {
         // Used to be indexer, now it is a computed property
         [x]: string;
         ~~~
@@ -40,10 +40,8 @@
         // Used to be indexer, now it is a computed property
         [x]: string
         ~~~
 !!! error TS1166: A computed property name in a class property declaration must have a simple literal type or a 'unique symbol' type.
-        ~~~
-!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
          ~
 !!! error TS2304: Cannot find name 'x'.
          ~
 !!! error TS4031: Public property '[x]' of exported class has or is using private name 'x'.
